sqlmesh/core/model/definition.py

```diff
@@ -1906,22 +1925,34 @@ def _split_sql_model_statements(
     Raises:
         ConfigError: If the model definition contains more than one SELECT query or `@INSERT_SEED()` call.
     """
+    from sqlmesh.core.audit.definition import load_audit, ModelAudit
+
     query_positions = []
-    for idx, expression in enumerate(expressions):
-        if (
+    inline_audits: UniqueKeyDict[str, ModelAudit] = UniqueKeyDict("inline_audits")
+    idx = 0
```

Comment:
why do you need idx manipulation?

Reply:
So that when an `Audit` instance is detected at the current index, to load it the corresponding next SQL statement and so that they are removed from the pre or post statement lists.

Reply:
This section is a bit complicated, I'd spend some more time thinking how it can be refactored.